### DISCUSSION TITLE
Does CodeQL scan aspx and ascx pages for C# and JS?

### DISCUSSION THREAD

**Discussion by flagrant99** [community/community#113463] [Actions] [answered]
### Select Topic Area

Question

### Body

CodeQL Tool Status lists C# (cs) and Javascript (js) files. 
I do not see any aspx or ascx. 
Does this means asp.net pages are omitted from CodeQL CSharp and Javascript scans?

**Comment by flagrant99** [comment]
I see evidence for CSharp scanning in aspx web pages. Not sure why these files are omitted from Tool Status. Still need to test javascript.

**Comment by tamasvajk** [comment]
~We do have some support for aspx files: when we build the database from the analyzed code, we're injecting `/p:MvcBuildViews=true` into the build command. This runs asp.net view file precompilation, and as a result we can include the compiled C# view files in the database. So we're not analysing directly aspx files, but rather we analyse the generated C# ones.~ 

Let me try answering again, the above covers how we analyze ASP.NET MVC applications. Some part of it applies to WebForms too, but it's not the default behaviour.

We do have some support for `.aspx` and `.ascx` files: the structure of these files are stored in the database, and it's exposed in QL through the classes here: https://github.com/github/codeql/blob/main/csharp/ql/lib/semmle/code/asp/AspNet.qll. Additionally, one could set up a build command in such a way that `.aspx` files are precompiled with `aspnet_compiler.exe`. Internally, this tool generates and compiles C# files, and these C# files can also be included in the database.

**Comment by sidshank** [comment]
I can confirm that the JavaScript code in ASCX pages is not extracted or scanned.

**Comment by flagrant99** [comment, marked as answer]
Thanks for the answers. So it seems JavaScript is not scanned at all in aspx, ascx files. However C# is.  [Tool Status](https://docs.github.com/en/code-security/code-scanning/managing-your-code-scanning-configuration/about-the-tool-status-page#how-codeql-defines-scanned-files) just doesn't list aspx, ascx files.
